File: Tests/installation_tests/manual_installation/ManualInstallationTest/MyAPIClient.m
//
//  MyAPIClient.m
//  ManualInstallationTest
//
//  Created by Ben Guo on 8/19/16.
//  Copyright © 2016 stripe. All rights reserved.
//

#import <Foundation/Foundation.h>
#import <Stripe/Stripe.h>
#import "MyAPIClient.h"

@interface MyAPIClient ()
@property (nonatomic, strong) STPCard *defaultSource;
@property (nonatomic, strong) NSArray<STPCard *> *sources;
@end

@implementation MyAPIClient

- (void)retrieveCustomer:(STPCustomerCompletionBlock)completion {
    STPCustomer *customer = [STPCustomer customerWithStripeID:@"cus_123" defaultSource:nil sources:@[]];
    completion(customer, nil);
}

- (void)attachSourceToCustomer:(id<STPSource>)source completion:(STPErrorBlock)completion {
    if ([source isKindOfClass:[STPToken class]]) {
        self.defaultSource = ((STPToken *)source).card;
    }
    completion(nil);
}

- (void)selectDefaultCustomerSource:(id<STPSource>)source completion:(STPErrorBlock)completion {
    if ([source isKindOfClass:[STPToken class]]) {
        self.defaultSource = ((STPToken *)source).card;
    }
    completion(nil);
}

@end

File: Tests/installation_tests/cocoapods/with_frameworks/CocoapodsTest/MyAPIClient.swift
//
//  MyAPIClient.swift
//  
//
//  Created by Ben Guo on 8/18/16.
//
//

import Foundation
import Stripe

class MyAPIClient: NSObject, STPBackendAPIAdapter {

    var defaultSource: STPCard? = nil
    var sources: [STPCard] = []

    @objc func retrieveCustomer(completion: STPCustomerCompletionBlock) {
        let customer = STPCustomer(stripeID: "cus_test", defaultSource: self.defaultSource, sources: self.sources)
        completion(customer, nil)
    }

    @objc func selectDefaultCustomerSource(source: STPSource, completion: STPErrorBlock) {
        if let token = source as? STPToken {
            self.defaultSource = token.card
        }
        completion(nil)
    }

    @objc func attachSourceToCustomer(source: STPSource, completion: STPErrorBlock) {
        if let token = source as? STPToken, card = token.card {
            self.sources.append(card)
            self.defaultSource = card
        }
        completion(nil)
    }
}
